### PR TITLE
PCHR-2153: Update modal window with new design and adding new fields

### PR DESCRIPTION
**Pull request by bmnepali**
## Overview
Elements like tabs (Details and Assignment) and form elements in New Document Modal needed to be reformatted and enhance for consistency and better UI/UX.

## Before
Refer screenshots in PR  #195, for screenshots of previous new document modal.

### A. Modal with Details Tab and  Assignment Tab respectively (gif)
new-document-iloveimg-compressed

## Technical Details
1. Since, the tabs were having the default bootstrap styles, hence were how consistent with current style guide, added some styles to maintain consistency of tabs in file `/scss/civitasks/partials/_nav-tabs.scss`
2. Reformatted and enhanced html markup of the New Document modal.

3. Backstopjs Tests:
Following failed test are due to the added changes as per new requirements, so they are fine.
screencapture-file-users-bmn-buildkit-build-hr16-new-sites-all-modules-civicrm-tools-extensions-civihr-uk-co-compucorp-civicrm-hrcore-backstop_data-html_report-index-html-1495035404108
- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss [hunk 1]
```diff
@@ -41,6 +41,12 @@ $input-border-radius:    0;
     border-top: 0 !important;
   }
 
+  .select2-chosen {
+    span {
+      color: $field-data-color;
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss]: `field-data-color` does not give any meaning or context , can you you more meaningful name that gives some more context to color and where its used?

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_nav-tabs.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Variables
+$inactive-tab:    #F4F7F8;
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_nav-tabs.scss]: how about `inactive-tab-light-blue-color` or `light-gray-color` that can be used at multiple places and not just in active tab and similarly for others

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_nav-tabs.scss [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+$active-tab:      #FFFFFF;
+
+$inactive-text:   #586277;
+$active-text:     #41AFCB;
+
+// Tabs
+.nav-tabs {
+  li {
+    background-color: $inactive-tab;
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_nav-tabs.scss]: why is it separated here whereas for `li.active` its part of a

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html [hunk 4]
```diff
@@ -34,6 +22,7 @@ <h4 class="modal-title">New Document</h4>
           </ui-select>
         </div>
       </div>
+
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: new line?

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html [hunk 5]
```diff
@@ -160,21 +169,26 @@ <h4 class="modal-title">New Document</h4>
...
-                  <div ng-bind="contact.label"></div>
-                  <small ng-bind="contact.description[0]"></small>
-                </ui-select-choices>
-              </ui-select>
-              <h4 ng-show="document.assignee_contact_id.length">Assignees:</h4>
+            <div class="col-xs-12 col-sm-12" ng-show="document.assignee_contact_id.length || showAssigneeField">
+              <div class="row">
+                <label class="col-xs-4 text-nowrap control-label"> Assignees: </label>
+                <div class="col-xs-8 input-group">
+                  <ui-select prevent-animations name="assignee" theme="civihr-ui-select" ng-model="document.temp" on-select="addAssignee($item);">
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: `temp` is not an informative name

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html [hunk 6]
```diff
@@ -218,9 +237,6 @@ <h4 ng-show="document.assignee_contact_id.length">Assignees:</h4>
     <button type="button" class="btn btn-default" ng-click="cancel()">
       Cancel
     </button>
-    <button type="submit" ng-click="openNew = true;" class="btn btn-default" ng-disabled="!documentForm.$valid">
-      Save And New
-    </button>
     <button type="submit" class="btn btn-default btn-primary">
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: is it not required to disable it when  `!documentForm.$valid`?

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss (hunk 1 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss]: ok!

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_nav-tabs.scss (hunk 2 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_nav-tabs.scss]: ok! will do

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html (hunk 5 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: Thanks for catching, will fix it.

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html (hunk 6 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: Right!, we don't need "Save and New Button" as per the requirement,  but it sounds sensible to disable a save button if the form is invalid. Will fix it.

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html (hunk 4 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: will fix it.

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_nav-tabs.scss (hunk 3 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_nav-tabs.scss]: will optimize it.

**Pull request merged**